jdaviz/configs/cubeviz/plugins/parsers.py

```diff
@@ -79,11 +83,13 @@ def parse_data(app, file_obj, data_type=None, data_label=None):
     # If the data types are custom data objects, use explicit parsers. Note
     #  that this relies on the glue-astronomy machinery to turn the data object
     #  into something glue can understand.
-    elif isinstance(file_obj, Spectrum1D):
+    elif isinstance(file_obj, Spectrum1D) and file_obj.flux.ndim in (1, 3):
         if file_obj.flux.ndim == 3:
             _parse_spectrum1d_3d(app, file_obj, data_label=data_label)
         else:
             _parse_spectrum1d(app, file_obj, data_label=data_label)
+    elif isinstance(file_obj, np.ndarray) and file_obj.ndim == 3:
```

Comment:
In a previous life over at #1040 , this code also would allow for `ndim=4` if the 4th dimension is only of size 1 (I heard some JWST data is weird like that) but supporting that is extra code, so since no one explicitly asked for that yet, I am excluding it in this PR to make review slightly faster.